lib/KBox/src/pages/NavigationPage.cpp

```diff
@@ -0,0 +1,280 @@
+                    apModeString = "APMode: Dodge     ";//extra spaces needed to overwrite old string if longer
+                    apModeDisplay->setText(apModeString);
+                    apModeDisplay->setColor(ColorRed);
+                }
+                return true;
+            } else {
+                DEBUG("apMode not active so no effect");
+            }
+        } else if (this->buttonPressedTimer > 5000) {  // Long click - start and stop the navMode here
+            DEBUG("long click");
+            if (imuCalibrated == true){
+                if (apMode == false){
+                    apMode = true;
+                    apHeadingMode = true;
+                    apModeString = "APMode: Heading       ";//extra spaces needed to overwrite old string if longer
+                    apModeDisplay->setText(apModeString);
+                    apModeDisplay->setColor(ColorGreen);
+                } else {
+                    apMode = false;
+                    apDodgeMode = false;
+                    apModeString = "APMode: Off           ";
+                    apModeDisplay->setText(apModeString);
+                    apModeDisplay->setColor(ColorBlue);
+                }
+            }
+            return true;
+        }
+    }
+    //DEBUG("APmode is: - %d", apMode);
+    //DEBUG("APHeadingmode is: - %d", apHeadingMode);
+    //DEBUG("APDodgemode is: - %d", apDodgeMode);
+    return true;
+}
+
+bool NavigationPage::processEvent(const EncoderEvent &ee) {
+    encoderClicks = ee.rotation;
+    navEncoderClicks = navEncoderClicks + encoderClicks;
+    encoderClicks = 0;
+    return true;
+}
+
+String NavigationPage::formatMeasurement(float measure, const char *unit) {
+  // extra spaces at the end needed to clear previous value completely
+  // (we use a non-fixed width font)
+  char s[10];
+  snprintf(s, sizeof(s), "%.0f %s  ", measure, unit); //do not display decimals for headings
+  return String(s);
+}
+
+
+void NavigationPage::processMessage(const KMessage &message) {
+  message.accept(*this);
+}
+
+void NavigationPage::visit(const APMessage &ap) {
+    navTargetRudderPosition = ap.getTargetRudderPosition();
+    navRudderCommandSent = ap.getRudderCommandSent();
+}
+
+void NavigationPage::visit(const RUDMessage &rm) {
+    navRudderSensorPosition = rm.getRudderDisplayAngle();
+}
+
+void NavigationPage::visit(const IMUMessage &imu) {
+    if (imu.getCalibration() == 3){
+        imuCalibrated = true;
+    }
+    if (imuCalibrated){
+        if (!apMode){
+            apModeString = "APMode: Off   ";
+            apModeDisplay->setText(apModeString);
+            apModeDisplay->setColor(ColorGreen);
+        }
+    } else {
+        apModeString = "Calibrating   ";
+        apModeDisplay->setText(apModeString);
+        apModeDisplay->setColor(ColorRed);
+    }
+    
+    rawHeading = RadToDeg(imu.getCourse());
+    if (rawHeading < 0){
+        navCurrentHeading = 720 + rawHeading;//adding 720 for nav math purposes
```

Comment:
So have thought a bit more about this.  I can see the logic for moving the currentHeading  / offCourse calculations into the autoPilotTask.  We still need to determine targetHeading in the navPage because it depends on encoder events that are only detected in the navPage.  So we still need to do the targetHeading math in the Nav page.  

I'm not sold on doing all the math in radians.  Everybody using this code is going to need to do some customization for their particular boat in determining what adjustments to use for MAXRUDDERSWING, deadzone, slack and the PID variables.  I don't think anybody will want to try and figure out what their MAXRUDDERSWING is in radians, or try and tune the AP performance based on radians.  I think that would be counterintuitive for just about everybody.  So I don't see any practical advantage to doing all the calculations in radians.  Am I missing something?

I'll go ahead and move the currentHeading calcs into the AutoPilot, but will hold off on doing everything in radians, pending a persuasive argument why that has practical, rather than purely theoretical benefits.